### ISSUE TITLE
[BUG] FP8+PP+Recompute+GA>1, loss = nan

### ISSUE THREAD

**Issue by jingjie01ai** [NVIDIA/Megatron-LM#607]
**Describe the bug**
FP8+PP+**Recompute**+GA>1, loss = nan
FP8+PP+GA>1, loss is normal
FP8+PP+Recompute+**GA=1**, loss is normal
FP8+**TP**+Recompute+GA>1, loss is normal

**Comment by jingjie01ai** [comment]
refer: https://github.com/NVIDIA/TransformerEngine/issues/539
